R/plot.SMCMallows.R

```diff
@@ -0,0 +1,127 @@
+#' @title Plot SMC Posterior Distributions
+#' @description Plot posterior distributions of SMC-Mallow parameters.
+#' @param x An object of type \code{SMC-Mallows}, returned for example from
+#' [smc_mallows_new_users()].
+#' @param nmc Number of Monte Carlo samples
+#' @param burnin A numeric value specifying the number of iterations
+#' to discard as burn-in. Defaults to \code{model_fit$burnin}, and must be
+#' provided if \code{model_fit$burnin} does not exist. See
+#' [assess_convergence()]
+#' @param parameter Character string defining the parameter to plot. Available
+#' options are \code{"alpha"} and \code{"rho"}.
+#' @param time Integer determining the update slice to plot
+#' @param C Number of cluster
+#' @param colnames A vector of item names. If NULL, generic names are generated
+#' for the items in the ranking.
+#' @param items Either a vector of item names, or a vector of indices. If NULL,
+#' five items are selected randomly.
+#' @param ... Other arguments passed to [base::plot()] (not used).
+#' @return A plot of the posterior distributions
+#' @author Waldir Leoncio
+#' @export
+#' @example /inst/examples/plot.SMCMallows_example.R
+plot.SMCMallows <- function(x, nmc = nrow(x$rho_samples[, 1, ]), burnin = 0,
+  parameter = "alpha", time = ncol(x$rho_samples[, 1, ]), C = 1,
+  colnames = NULL, items = NULL, ...) {
+
+  if (parameter == "alpha") {
+    output <- x$alpha_samples[, time]
+    plot_alpha_posterior(output, nmc, burnin)
+  } else if (parameter == "rho") {
+    output <- x$rho_samples[, , time]
+    plot_rho_posterior(output, nmc, burnin, C, colnames, items)
+  } else {
+    stop("parameter must be either 'alpha' or 'rho'.")
+  }
+}
+
+#' @title Plot Alpha Posterior
+#' @description posterior for alpha
+#' @inheritParams compute_posterior_intervals_rho
+#' @export
```

Comment:
@wleoncio, wouldn't it be best to not export plot_alpha_posterior and plot_rho_posterior now that they are being called by plot.SMCMallows?

Reply:
It would definitely be cleaner and probably clearer for the user, but I wonder if we should take a mid-step before completely dropping it, considering how many SMC functions are already listed in [smc_mallows_deprecated.R](https://github.com/ocbe-uio/BayesMallows/blob/036710dcc43d06319118e7c3a05f8c5070a0184d/R/smc_mallows_deprecated.R). My suggestion would be to:

1. Add `plot_alpha_posterior()` and `plot_rho_posterior()` to smc_mallows_deprecated.R
2. Rename and internalize the subfunctions in plot.SMCMallows.R to avoid conflicts (e.g. `plot_smc_alpha()` and `plot_smc_rho()`
3. Eventually release a new BayesMallows version completely removing support for all deprecated functions. Even though we're not strictly following SemVer here, [what they have to say about handling deprecating functions](https://semver.org/#how-should-i-handle-deprecating-functionality) sounds like a good plan overall

How does that sound?

Reply:
> https://github.com/ocbe-uio/BayesMallows/pull/263#issuecomment-1328801617

OK, I'll open a PR implementing points 1 and 2 above, then we deal with #3 later. :)